Seems like the wrong SDL version is vendored.

Just wanted to play around with your library. However after installing the dependencies with glide and trying to build the examples I see a lot of errors

```
# github.com/tanema/amore/vendor/github.com/tanema/amore/joystick
../../amore/vendor/github.com/tanema/amore/joystick/joystick.go:36:50: cannot use sdl.JoystickID(joystick.id) (type sdl.JoystickID) as type int in argument to sdl.JoystickOpen
../../amore/vendor/github.com/tanema/amore/joystick/joystick.go:68:48: joystick.stick.GetAttached undefined (type *sdl.Joystick has no field or method GetAttached)
../../amore/vendor/github.com/tanema/amore/joystick/joystick.go:73:49: joystick.stick.GetGUID undefined (type *sdl.Joystick has no field or method GetGUID)
../../amore/vendor/github.com/tanema/amore/joystick/joystick.go:98:23: joystick.stick.GetButton undefined (type *sdl.Joystick has no field or method GetButton)
../../amore/vendor/github.com/tanema/amore/joystick/joystick.go:131:40: joystick.stick.GetAxis undefined (type *sdl.Joystick has no field or method GetAxis)
../../amore/vendor/github.com/tanema/amore/joystick/joystick.go:144:54: joystick.stick.GetAxis undefined (type *sdl.Joystick has no field or method GetAxis)
../../amore/vendor/github.com/tanema/amore/joystick/joystick.go:156:23: joystick.stick.GetHat undefined (type *sdl.Joystick has no field or method GetHat)
../../amore/vendor/github.com/tanema/amore/joystick/joystick.go:165:30: joystick.controller.GetAxis undefined (type *sdl.GameController has no field or method GetAxis)
../../amore/vendor/github.com/tanema/amore/joystick/joystick.go:176:28: joystick.controller.GetButton undefined (type *sdl.GameController has no field or method GetButton)
../../amore/vendor/github.com/tanema/amore/joystick/joystick.go:251:58: cannot use &joystick.vibration.Effect (type *sdl.HapticEffect) as type sdl.iHapticEffect in argument to joystick.haptic.UpdateEffect:
	*sdl.HapticEffect does not implement sdl.iHapticEffect (missing sdl.cHapticEffect method)
../../amore/vendor/github.com/tanema/amore/joystick/joystick.go:251:58: too many errors
# github.com/tanema/amore/vendor/github.com/tanema/amore/window
../../amore/vendor/github.com/tanema/amore/window/window.go:63:28: too many arguments in call to sdl.GetDesktopDisplayMode
	have (int, *sdl.DisplayMode)
	want (int)
../../amore/vendor/github.com/tanema/amore/window/window.go:83:33: too many arguments in call to sdl.GetDisplayMode
	have (int, number, *sdl.DisplayMode)
	want (int, int)
../../amore/vendor/github.com/tanema/amore/window/window.go:83:33: multiple-value sdl.GetDisplayMode() in single-value context
../../amore/vendor/github.com/tanema/amore/window/window.go:108:23: too many arguments in call to sdl.GetDisplayBounds
	have (int, *sdl.Rect)
	want (int)
../../amore/vendor/github.com/tanema/amore/window/window.go:167:18: undefined: sdl.GLCreateContext
../../amore/vendor/github.com/tanema/amore/window/window.go:244:28: too many arguments in call to sdl.GetDesktopDisplayMode
	have (int, *sdl.DisplayMode)
	want (int)
../../amore/vendor/github.com/tanema/amore/window/window.go:261:10: undefined: sdl.GLGetDrawableSize
../../amore/vendor/github.com/tanema/amore/window/window.go:271:21: too many arguments in call to sdl.GetDisplayMode
	have (int, int, *sdl.DisplayMode)
	want (int, int)
../../amore/vendor/github.com/tanema/amore/window/window.go:332:2: undefined: sdl.GLSwapWindow
../../amore/vendor/github.com/tanema/amore/window/window.go:395:23: multiple-value sdl.ShowCursor() in single-value context
../../amore/vendor/github.com/tanema/amore/window/window.go:395:23: too many errors
```

My guess is that the wrong version of go-sdl2 is vendored?

--- a/audio/decoding/mp3.go
+++ b/audio/decoding/mp3.go
@@ -7,7 +7,7 @@ import (
 )
 
 func newMp3Decoder(src io.ReadCloser) (*Decoder, error) {
-	r, err := mp3.Decode(src)
+	r, err := mp3.NewDecoder(src)
 	if err != nil {
 		return nil, err
 	}

--- a/glide.lock
+++ b/glide.lock
@@ -1,5 +1,5 @@
-hash: 603f577420c77a8e6d91e57f33e0b778419476db89947888267817a236ebfa71
-updated: 2017-12-18T21:59:43.378156-05:00
+hash: 7f809f65d66cbdf9a78ac7e817277cb59d12b2d40214bfa1755d8ca95faa8608
+updated: 2018-05-03T09:47:20.162743-04:00
 imports:
 - name: github.com/BurntSushi/toml
   version: b26d9c308763d68093482582cea63d69be07a0f0
@@ -8,11 +8,11 @@ imports:
 - name: github.com/eaburns/flac
   version: 9a6fb92396d1ba6412b82819435dca0b46f959fb
 - name: github.com/go-gl/gl
-  version: ac0d3d2af0fe995e2bb09fa6619b716b0c0fc0fa
+  version: 68e2537930806bfcee1b92a0e14b4e9b8bb3a3e3
   subpackages:
   - v2.1/gl
 - name: github.com/go-gl/mathgl
-  version: 4c3fc6b4bf30179013667e5d0b926e024d9a13c1
+  version: 5ab0e04e1f55a10f3a71b1fccc60ca2ee9a0bc02
   subpackages:
   - mgl32
   - mgl32/matstack
@@ -22,7 +22,7 @@ imports:
   - raster
   - truetype
 - name: github.com/hajimehoshi/go-mp3
-  version: 551b228a29684c056919369a0c6516a63bb943e8
+  version: 3c185f92b8dbceefa913b64cae634ba47f452769
   subpackages:
   - internal/bits
   - internal/consts
@@ -40,23 +40,21 @@ imports:
   version: 75fb7ed4208cf72d323d7d02fd1a5964a7a9073c
   subpackages:
   - open
-- name: github.com/tanema/freetype-go
-  version: dc1e2b83a6a63e7740967c29ad64b3de5ed93495
-  subpackages:
-  - freetype
-  - freetype/truetype
 - name: github.com/veandco/go-sdl2
-  version: d1ebfcf49872383b6533ee2a51ae478ae3999c0c
+  version: 30f30965227962032c13bd0b12a89e6a5107c768
   subpackages:
   - sdl
 - name: golang.org/x/image
-  version: 12117c17ca67ffa1ce22e9409f3b0b0a93ac08c7
+  version: f315e440302883054d0c2bd85486878cb4f8572c
   subpackages:
   - font
+  - font/gofont/gobold
+  - font/gofont/goitalic
+  - font/gofont/goregular
   - math/f32
   - math/fixed
 - name: golang.org/x/mobile
-  version: 50b7067763308074f4c19e733ac7f867471de857
+  version: c909788f9903c3fa409bdba28f909907d69fb3d1
   subpackages:
   - internal/mobileinit
 testImports: []

--- a/glide.yaml
+++ b/glide.yaml
@@ -1,7 +1,7 @@
 package: github.com/tanema/amore
 import:
 - package: github.com/BurntSushi/toml
-  version: ^0.2.0
+  version: ^0.3.0
 - package: github.com/eaburns/flac
 - package: github.com/go-gl/gl
   subpackages:
@@ -10,19 +10,26 @@ import:
   subpackages:
   - mgl32
   - mgl32/matstack
+- package: github.com/golang/freetype
+  subpackages:
+  - truetype
+- package: github.com/hajimehoshi/go-mp3
+- package: github.com/jfreymuth/oggvorbis
+  version: ^1.0.0
 - package: github.com/skratchdot/open-golang
   subpackages:
   - open
-- package: github.com/tanema/freetype-go
-  subpackages:
-  - freetype
-  - freetype/truetype
 - package: github.com/veandco/go-sdl2
+  version: ^0.2.0
   subpackages:
   - sdl
+- package: golang.org/x/image
+  subpackages:
+  - font
+  - font/gofont/gobold
+  - font/gofont/goitalic
+  - font/gofont/goregular
+  - math/fixed
 - package: golang.org/x/mobile
   subpackages:
   - internal/mobileinit
-- package: github.com/jfreymuth/oggvorbis
-- package: github.com/golang/freetype
-- package: github.com/hajimehoshi/go-mp3

--- a/joystick/joystick.go
+++ b/joystick/joystick.go
@@ -11,10 +11,8 @@ import (
 // vibration is a struct that keeps track of vibration patterns and lengths for
 // communication with sdl
 type vibration struct {
-	Left, Right float32
-	Effect      sdl.HapticEffect
-	Data        [4]uint16
 	ID          int
+	Left, Right float32
 	Endtime     uint32
 }
 
@@ -33,7 +31,7 @@ type Joystick struct {
 func (joystick *Joystick) open() bool {
 	joystick.close()
 
-	joystick.stick = sdl.JoystickOpen(sdl.JoystickID(joystick.id))
+	joystick.stick = sdl.JoystickOpen(joystick.id)
 
 	if joystick.stick != nil && sdl.IsGameController(joystick.id) {
 		joystick.controller = sdl.GameControllerOpen(joystick.id)
@@ -65,12 +63,12 @@ func (joystick *Joystick) GetName() string {
 
 // IsConnected gets whether the Joystick is connected.
 func (joystick *Joystick) IsConnected() bool {
-	return joystick.stick != nil && joystick.stick.GetAttached()
+	return joystick.stick != nil && joystick.stick.Attached()
 }
 
 // GetGUID gets a stable GUID unique to the type of the physical joystick.
 func (joystick *Joystick) GetGUID() string {
-	return sdl.JoystickGetGUIDString(joystick.stick.GetGUID())
+	return sdl.JoystickGetGUIDString(joystick.stick.GUID())
 }
 
 // getHandle gets the sdl instance id for this joystick
@@ -95,7 +93,7 @@ func (joystick *Joystick) IsDown(button int) bool {
 	if joystick.IsConnected() == false {
 		return false
 	}
-	return joystick.stick.GetButton(button) == 1
+	return joystick.stick.Button(button) == 1
 }
 
 // GetAxisCount gets the number of axes on the joystick.
@@ -128,7 +126,7 @@ func (joystick *Joystick) GetAxis(axisindex int) float32 {
 		return 0.0
 	}
 
-	return clampval(float32(joystick.stick.GetAxis(axisindex)) / 32768.0)
+	return clampval(float32(joystick.stick.Axis(axisindex)) / 32768.0)
 }
 
 // GetAxes gets the direction of each axis. Values are clamped.
@@ -141,7 +139,7 @@ func (joystick *Joystick) GetAxes() []float32 {
 	}
 
 	for i := 0; i < count; i++ {
-		axes = append(axes, clampval(float32(joystick.stick.GetAxis(i))/32768.0))
+		axes = append(axes, clampval(float32(joystick.stick.Axis(i))/32768.0))
 	}
 
 	return axes
@@ -153,7 +151,7 @@ func (joystick *Joystick) GetHat(hatindex int) byte {
 		return 0
 	}
 
-	return joystick.stick.GetHat(hatindex)
+	return joystick.stick.Hat(hatindex)
 }
 
 // GetGamepadAxis gets the direction of a virtual gamepad axis. Values are clamped.
@@ -162,7 +160,7 @@ func (joystick *Joystick) GetGamepadAxis(axis GameControllerAxis) float32 {
 		return 0.0
 	}
 
-	value := joystick.controller.GetAxis(sdl.GameControllerAxis(axis))
+	value := joystick.controller.Axis(sdl.GameControllerAxis(axis))
 
 	return clampval(float32(value) / 32768.0)
 }
@@ -173,7 +171,7 @@ func (joystick *Joystick) IsGamepadDown(button GameControllerButton) bool {
 		return false
 	}
 
-	return joystick.controller.GetButton(sdl.GameControllerButton(button)) == 1
+	return joystick.controller.Button(sdl.GameControllerButton(button)) == 1
 }
 
 // IsVibrationSupported gets whether the Joystick supports vibration.
@@ -233,22 +231,16 @@ func (joystick *Joystick) checkCreateHaptic() bool {
 		return false
 	}
 
-	joystick.vibration = &vibration{
-		ID:      -1,
-		Left:    0.0,
-		Right:   0.0,
-		Endtime: sdl.HAPTIC_INFINITY,
-		Effect:  sdl.HapticEffect{},
-	}
+	joystick.vibration = &vibration{ID: -1, Endtime: sdl.HAPTIC_INFINITY}
 
 	return joystick.haptic != nil
 }
 
 // runVibrationEffect will initiate the joysticks haptic vibration and will return
 // if it was successful in running the vibration.
-func (joystick *Joystick) runVibrationEffect() bool {
+func (joystick *Joystick) runVibrationEffect(effect sdl.HapticEffect) bool {
 	if joystick.vibration.ID != -1 {
-		if joystick.haptic.UpdateEffect(joystick.vibration.ID, &joystick.vibration.Effect) == nil {
+		if joystick.haptic.UpdateEffect(joystick.vibration.ID, effect) == nil {
 			if joystick.haptic.RunEffect(joystick.vibration.ID, 1) == nil {
 				return true
 			}
@@ -260,7 +252,7 @@ func (joystick *Joystick) runVibrationEffect() bool {
 	}
 
 	var err error
-	joystick.vibration.ID, err = joystick.haptic.NewEffect(&joystick.vibration.Effect)
+	joystick.vibration.ID, err = joystick.haptic.NewEffect(effect)
 	if err == nil && joystick.vibration.ID != -1 && joystick.haptic.RunEffect(joystick.vibration.ID, 1) == nil {
 		return true
 	}
@@ -312,12 +304,12 @@ func (joystick *Joystick) SetVibration(args ...float32) bool {
 	}
 
 	if (features & sdl.HAPTIC_LEFTRIGHT) != 0 {
-		joystick.vibration.Effect.SetType(sdl.HAPTIC_LEFTRIGHT)
-		lr := joystick.vibration.Effect.LeftRight()
-		lr.Length = uint32(length)
-		lr.LargeMagnitude = uint16(left * math.MaxUint16)
-		lr.SmallMagnitude = uint16(right * math.MaxUint16)
-		success = joystick.runVibrationEffect()
+		success = joystick.runVibrationEffect(&sdl.HapticLeftRight{
+			Type:           sdl.HAPTIC_LEFTRIGHT,
+			Length:         uint32(length),
+			LargeMagnitude: uint16(left * math.MaxUint16),
+			SmallMagnitude: uint16(right * math.MaxUint16),
+		})
 	}
 
 	// Some gamepad drivers only give support for controlling individual motors
@@ -327,32 +319,33 @@ func (joystick *Joystick) SetVibration(args ...float32) bool {
 		// but aren't similar to https://github.com/d235j/360Controller .
 
 		// Custom effect data is clamped to 0x7FFF in SDL.
-		joystick.vibration.Data[0] = uint16(left * 0x7FFF)
-		joystick.vibration.Data[2] = uint16(left * 0x7FFF)
-		joystick.vibration.Data[1] = uint16(right * 0x7FFF)
-		joystick.vibration.Data[3] = uint16(right * 0x7FFF)
+		data := []uint16{
+			uint16(left * 0x7FFF),
+			uint16(left * 0x7FFF),
+			uint16(right * 0x7FFF),
+			uint16(right * 0x7FFF),
+		}
 
-		joystick.vibration.Effect.SetType(sdl.HAPTIC_CUSTOM)
-		custom := joystick.vibration.Effect.Custom()
-		custom.Length = uint32(length)
-		custom.Channels = 2
-		custom.Period = 10
-		custom.Samples = 2
-		custom.Data = &joystick.vibration.Data[0]
-
-		success = joystick.runVibrationEffect()
+		success = joystick.runVibrationEffect(&sdl.HapticCustom{
+			Type:     sdl.HAPTIC_CUSTOM,
+			Length:   uint32(length),
+			Channels: 2,
+			Period:   10,
+			Samples:  2,
+			Data:     &data[0],
+		})
 	}
 
 	//// Fall back to a simple sine wave if all else fails. This only supports a
 	//// single strength value.
 	if !success && (features&sdl.HAPTIC_SINE) != 0 {
-		joystick.vibration.Effect.SetType(sdl.HAPTIC_SINE)
-		periodic := joystick.vibration.Effect.Periodic()
-		periodic.Length = uint32(length)
-		periodic.Period = 10
 		strength := math.Max(float64(left), float64(right))
-		periodic.Magnitude = int16(strength * 0x7FFF)
-		success = joystick.runVibrationEffect()
+		success = joystick.runVibrationEffect(&sdl.HapticPeriodic{
+			Type:      sdl.HAPTIC_SINE,
+			Length:    uint32(length),
+			Period:    10,
+			Magnitude: int16(strength * 0x7FFF),
+		})
 	}
 
 	if success {
@@ -405,12 +398,10 @@ func (joystick *Joystick) GetVibration() (float32, float32) {
 	}
 
 	// Check if the haptic effect has stopped playing.
-	if joystick.haptic == nil || joystick.vibration.ID == -1 {
-		status, err := joystick.haptic.GetEffectStatus(joystick.vibration.ID)
-		if err == nil && status != 1 {
-			joystick.vibration.Left = 0.0
-			joystick.vibration.Right = 0.0
-		}
+	status, err := joystick.haptic.GetEffectStatus(joystick.vibration.ID)
+	if err == nil && status != 1 {
+		joystick.vibration.Left = 0.0
+		joystick.vibration.Right = 0.0
 	}
 
 	return joystick.vibration.Left, joystick.vibration.Right

--- a/mouse/mouse.go
+++ b/mouse/mouse.go
@@ -76,7 +76,8 @@ func SetGrabbed(enabled bool) {
 
 // IsVisible checks if the cursor is visible.
 func IsVisible() bool {
-	return sdl.ShowCursor(sdl.QUERY) == sdl.ENABLE
+	status, err := sdl.ShowCursor(sdl.QUERY)
+	return status == sdl.ENABLE && err == nil
 }
 
 // SetVisible sets the current visibility of the cursor.

--- a/window/window.go
+++ b/window/window.go
@@ -59,8 +59,7 @@ func NewWindow() (*window, error) {
 	config.Display = int(math.Min(math.Max(float64(config.Display), 0.0), float64(GetDisplayCount()-1)))
 
 	if config.Width == 0 || config.Height == 0 {
-		var mode sdl.DisplayMode
-		sdl.GetDesktopDisplayMode(config.Display, &mode)
+		mode, _ := sdl.GetDesktopDisplayMode(config.Display)
 		config.Width = mode.W
 		config.Height = mode.H
 	}
@@ -80,7 +79,7 @@ func NewWindow() (*window, error) {
 				// GetClosestDisplayMode will fail if we request a size larger
 				// than the largest available display mode, so we'll try to use
 				// the largest (first) mode in that case.
-				if err := sdl.GetDisplayMode(config.Display, 0, &mode); err != nil {
+				if mode, err = sdl.GetDisplayMode(config.Display, 0); err != nil {
 					return nil, err
 				}
 			}
@@ -104,8 +103,7 @@ func NewWindow() (*window, error) {
 
 	if config.Fullscreen {
 		// The position needs to be in the global coordinate space.
-		var displaybounds sdl.Rect
-		sdl.GetDisplayBounds(config.Display, &displaybounds)
+		displaybounds, _ := sdl.GetDisplayBounds(config.Display)
 		config.X += displaybounds.X
 		config.Y += displaybounds.Y
 	} else {
@@ -164,7 +162,7 @@ func createWindowAndContext(config *windowConfig, windowflags uint32) (*window, 
 		panic(err)
 	}
 
-	context, err := sdl.GLCreateContext(newWindow)
+	context, err := newWindow.GLCreateContext()
 	if err != nil {
 		panic(err)
 	}
@@ -240,8 +238,7 @@ func GetDisplayName(displayindex int) string {
 func GetDesktopDimensions(displayindex int) (int32, int32) {
 	var width, height int32
 	if displayindex >= 0 && displayindex < GetDisplayCount() {
-		var mode sdl.DisplayMode
-		sdl.GetDesktopDisplayMode(displayindex, &mode)
+		mode, _ := sdl.GetDesktopDisplayMode(displayindex)
 		width = mode.W
 		height = mode.H
 	}
@@ -258,7 +255,7 @@ func onSizeChanged(width, height int32) {
 
 // GetDrawableSize gets the size of pixels in view
 func GetDrawableSize() (int32, int32) {
-	w, h := sdl.GLGetDrawableSize(getCurrent().SDLWindow)
+	w, h := getCurrent().SDLWindow.GLGetDrawableSize()
 	return int32(w), int32(h)
 }
 
@@ -267,8 +264,7 @@ func GetFullscreenSizes(displayindex int) [][]int32 {
 	var sizes [][]int32
 	modes, _ := sdl.GetNumDisplayModes(displayindex)
 	for i := 0; i < modes; i++ {
-		var mode sdl.DisplayMode
-		sdl.GetDisplayMode(displayindex, i, &mode)
+		mode, _ := sdl.GetDisplayMode(displayindex, i)
 		sizes = append(sizes, []int32{mode.W, mode.H})
 	}
 	return sizes
@@ -329,7 +325,7 @@ func Close(shouldClose bool) {
 // SwapBuffers swaps the current frame buffer in the window. This is generally
 // only used by the engine but you could use it to roll your own game loop.
 func SwapBuffers() {
-	sdl.GLSwapWindow(getCurrent().SDLWindow)
+	getCurrent().SDLWindow.GLSwap()
 }
 
 // ToPixelCoords translates window coords to pixel coords for pixel perfect
@@ -392,7 +388,8 @@ func SetMouseVisible(visible bool) {
 // false otherwise
 func GetMouseVisible() bool {
 	getCurrent() // must call getCurrent to ensure there is a window
-	return sdl.ShowCursor(sdl.QUERY) == sdl.ENABLE
+	status, err := sdl.ShowCursor(sdl.QUERY)
+	return status == sdl.ENABLE && err == nil
 }
 
 // GetPixelDimensions will return the size of the window in pixels

